spirv_cross/src/wrapper.cpp

```diff
@@ -65,6 +65,28 @@ ScInternalResult sc_internal_compiler_msl_new(ScInternalCompilerMsl **compiler,
     INTERNAL_RESULT(*compiler = new spirv_cross::CompilerMSL(ir, size);)
 }
 
+ScInternalResult sc_internal_compiler_msl_compile(const ScInternalCompilerBase *compiler, const char **shader,
+                                                  const spirv_cross::MSLVertexAttr *p_vat_overrides, size_t vat_override_count,
+                                                  const spirv_cross::MSLResourceBinding *p_res_overrides, size_t res_override_count)
+{
+    INTERNAL_RESULT(
+        do {
+            std::vector<spirv_cross::MSLVertexAttr> vat_overrides;
+            if (p_vat_overrides)
+            {
+                vat_overrides.insert(vat_overrides.end(), &p_vat_overrides[0], &p_vat_overrides[vat_override_count]);
```

Comment:
I don't understand why we need to copy out the overrides in an `std::vector`, given that it has the same (or even smaller) lifetime than the original?

Reply:
The lifetime is fine without this copying, it's the conversion to `std::vector` to match the [`compile` signature which requires a `std::vector`](https://github.com/KhronosGroup/SPIRV-Cross/blob/master/spirv_msl.cpp#L143) (unlike the constructor overloads which [support](https://github.com/KhronosGroup/SPIRV-Cross/blob/master/spirv_msl.cpp#L30) [either](https://github.com/KhronosGroup/SPIRV-Cross/blob/master/spirv_msl.cpp#L46)). The memory storage is the same though, so maybe there is a way to wrap the struct array directly into `std::vector`?

Reply:
Oh, I see. Sucks... but hey, at least it's an internal detail.
I'll try to ask around about the std::vector next week and we'll revise it.

Reply:
Ok, answer is - not possible ATM.
We need to make PR to spirv-cross to add an overload without `std::vector`.

Reply:
Ah, ok. I'll try do that.